extensions/wikia/CommunityMessages/CommunityMessages.class.php

```diff
@@ -114,10 +114,33 @@ static function onArticleSaveComplete( WikiPage $article, User $user, $text, $su
 	static function onBeforePageDisplay( OutputPage $out, Skin $skin ): bool {
 		$title = $out->getTitle();
 
-		if ( $title->isSpecial('ActivityFeed') || $title->isSpecial('MyHome') || $title->isSpecial('WikiActivity') ) {
-			self::dismissMessage();
+		if ( !$title->isSpecial( 'ActivityFeed' ) && !$title->isSpecial( 'MyHome' ) &&
+			 !$title->isSpecial( 'WikiActivity' ) ) {
+			return true;
+		}
+
+		global $wgMemc;
+		$communityMessagesTimestamp = $wgMemc->get( static::getCommunityMessagesCacheKey() ) ?? time();
+
+		// SUS-2585: Update table for logged in users
+		$user = $out->getUser();
+		if ( $user->isLoggedIn() && $communityMessagesTimestamp > (int) static::getUserTimestamp( $user ) ) {
+			global $wgCityId;
+
+			$task = ( new \Wikia\Tasks\Tasks\DismissCommunityMessageTask() )
+				->wikiId( $wgCityId )
+				->createdBy( $user );
+
+			$task->call( 'dismissCommunityMessage', $communityMessagesTimestamp );
+			$task->queue();
+
+			return true;
 		}
 
+		// set cookie for anons
+		$out->getRequest()->response()
+			->setcookie('CommunityMessages', $communityMessagesTimestamp, time() + 86400 /*24h*/);
```

Comment:
But logged-in user can still reach this line (when the timestamps do not match in the previous if statement).